src/main/kotlin/com/jobtracker/api/controllers/ApplicationController.kt

```diff
@@ -80,4 +80,29 @@ class ApplicationController(
         applicationRepository.deleteById(UUID.fromString(appId))
         return ResponseEntity.status(HttpStatus.NO_CONTENT).body(null)
     }
+
+    @PutMapping("/applications/{appId}")
+    fun editApplication(
+        @PathVariable appId: String,
+
+        @RequestBody application: ApplicationModel,
+        @RequestHeader("Authorization") token: String):ResponseEntity<Any> {
+
+        val user = Helpers.getUserByJWT(token, jwtDecoder, userRepository)
+            ?: return ResponseEntity.status(HttpStatus.NOT_FOUND).body(ErrorModel(404, "Company with ID does not exist"))
```

Comment:
Should the error message for this be related to the user? It looks like the line above is trying to get the User entity using their JWT.